dFusion/dFusion.rst

```diff
@@ -184,170 +182,230 @@ TODO
 
-All orders are encoded as limit sell orders: **(accountId, fromTokenIndex, toTokenIndex, limitPrice, amount, batchId, signature)**.
-The order should be read in the following way: the user occupying the specified *accountId* would like to sell the token *fromTokenIndex* for *toTokenIndex* for at most the *limitPrice* and the *amount* specified.
-The *batchId* and *signature* allow a third party to submit an order on behalf of others (saving gas when batching multiple orders together).
-The user only has to specify which batch their order is valid for and sign all the information with their private key.
+All orders are encoded as limit sell orders: **(accountId, fromTokenIndex, toTokenIndex, buyAmount, sellAmount, validUntilAuctionId, flagIsBuy, flagIsCancelation, signature)**.
+The order should be read in the following way: the user occupying the specified *accountId* would like to sell the token *fromTokenIndex* for *toTokenIndex* for at most the ratio *buyAmount* / *sellAmount*.
+Additionally, the user would like to buy at most *buyAmount* tokens if the *flag_isBuy* is true, otherwise, he would like to sell at most *sellAmount* tokens.
+Any placed order is placed into an order stream, a queue data type.
+Any order in the orderstream is valid until the auction with the id *validUntilAuctionId* is reached or the order is popped out of the queue data, due to a new insert.
+Additionally, an order can be resubmitted with the positive *flagIsCancelation* and then the order is also considered to be canceled. 
+It does not matter, whether the cancelation order is before or after the actual order, in any case, the order is canceled.
+If we would not have this logic, then anyone could just replay canceled orders.
+
+The order stream is queue able to hold 2**24 placed orders or order cancelations.
+The order stream is finite, as any solutions need to index orders with a certain amount of bits (24).
+Orders in the order stream are batched into smaller batches of size 2**7, and for each of these batches, the rolling order hash is stored on-chain.
+Each solution will just be able to consider the orders from the order stream stored in the last 2**(24-7) rolling order batches.
+
+The *signature* must sign the order with the private key of the *accountID*.
 
 The anchor smart contract on ethereum will offer the following function:
 
 .. code:: js
 
-    function appendOrders(bytes32 [] orders) { 
+    function appendOrders(bytes [] orders) { 
         // some preliminary checks limiting the number of orders..
 
         // update of orderHashSha
         for(i=0; i<orders.length; i++){
             if("check signature and batchID of order") {
                 // hash order without signature
-                byte32 oldHashSha = orderHashSha
-                orderHashSha = Kecca256(oldHashSha, orders[i]) 
-                emit OrderSubmitted(oldHashSha, orders[i], orderHashSha)
+                byte32 oldHashSha = orderHashSha[orderBatchCount]
```

Comment:
what is `orderBatchCount`? Is this `auctionId`?

Reply:
Order stream is made of several orderBatches. Here we just have a counter for the orderbatches